cpp/src/io/statistics/statistics_type_identification.cuh

```diff
@@ -174,8 +175,10 @@ class aggregation_type {
   using integral_aggregation_type =
     typename std::conditional_t<std::is_signed_v<T>, int64_t, uint64_t>;
 
-  using arithmetic_aggregation_type =
-    typename std::conditional_t<std::is_integral_v<T>, integral_aggregation_type, double>;
+  using arithmetic_aggregation_type = typename std::conditional_t<
+    std::is_integral_v<T>,
+    std::conditional_t<std::is_same_v<T, __int128_t>, __int128_t, integral_aggregation_type>,
+    double>;
 
   using non_arithmetic_aggregation_type =
     typename std::conditional_t<cudf::is_fixed_point<T>() or cudf::is_duration<T>() or
```

Comment:
decimal128 is a fixed point type so it'd be better to modify this.

Reply:
That's what I thought initially, but it seems that by the time a`ggregation_type` is involved, the decimal128 has already be turned into the int128 representation.  When I modified `non_arithmetic_aggregation_type` as I did above in `extrema_type` it would not even compile.

Reply:
I'm very confused.  I think part of the problem is at typed_statistics_chunk.cuh line 136.  
```c++
aggregate += detail::aggregation_type<A>::convert(union_member::get<A>(chunk.sum));
```
It seems like the A should be T.  For T=decimal128, A would be __int128_t, so calling aggregation_type::convert with T=__int128_t fails.  But if I change A to T, I get different errors.

Reply:
@devavret I reverted arithmetic_aggregation_type and changed non_arithmetic_aggregation_type as you suggested.  To get this to work, I changed typed_statistics_chunk.cuh line 136 to get rid of the convert call.  since `get<A>(chunk.sum)` already returns the aggregation type, there's no need to call convert.

@vuule can you look at this too and see if you think it will work for ORC statistics?

Reply:
The changes look good to me. There's no immediate impact on ORC since we don't support decimal statistics in ORC yet.

Reply:
Looks good now.